components/modbus_controller.rst

```diff
@@ -117,6 +117,90 @@ Technically there is no difference between the "inline" and the standard definit
         value_type: U_WORD
 
 
+Bitmasks
+--------
+
+Some devices use decimal values in read registers to show multiple binary states occupying only one register address. To decode them, you can use bitmasks according to the table below. The decimal value corresponding to a bit is always double of the previous one in the row. Multiple bits can be represented in a single register by making a sum of all the values corresponding to the bits.
+
++------------+------------------+-----------+-----------+ 
+| Alarm  bit | Description      | DEC value | HEX value |
++============+==================+===========+===========+ 
+| bit 0      | Binary Sensor 0  | 1         | 1         |
++------------+------------------+-----------+-----------+ 
+| bit 1      | Binary Sensor 1  | 2         | 2         |
++------------+------------------+-----------+-----------+ 
+| bit 2      | Binary Sensor 2  | 4         | 4         |
++------------+------------------+-----------+-----------+ 
+| bit 3      | Binary Sensor 3  | 8         | 8         |
++------------+------------------+-----------+-----------+ 
+| bit 4      | Binary Sensor 4  | 16        | 10        |
++------------+------------------+-----------+-----------+ 
+| bit 5      | Binary Sensor 5  | 32        | 20        |
++------------+------------------+-----------+-----------+ 
+| bit 6      | Binary Sensor 6  | 64        | 40        |
++------------+------------------+-----------+-----------+ 
+| bit 7      | Binary Sensor 7  | 128       | 80        |
++------------+------------------+-----------+-----------+ 
+| bit 8      | Binary Sensor 8  | 256       | 100       |
++------------+------------------+-----------+-----------+ 
+| bit 9      | Binary Sensor 9  | 512       | 200       |
++------------+------------------+-----------+-----------+ 
+| bit 10     | Binary Sensor 10 | 1024      | 400       |
++------------+------------------+-----------+-----------+ 
+| bit 11     | Binary Sensor 11 | 2048      | 800       |
++------------+------------------+-----------+-----------+ 
+| bit 12     | Binary Sensor 12 | 4096      | 1000      |
++------------+------------------+-----------+-----------+ 
+| bit 13     | Binary Sensor 13 | 8192      | 2000      |
++------------+------------------+-----------+-----------+ 
+| bit 14     | Binary Sensor 14 | 16384     | 4000      |
++------------+------------------+-----------+-----------+ 
+| bit 15     | Binary Sensor 15 | 32768     | 8000      |
++------------+------------------+-----------+-----------+ 
+
+For example, at read register ``15``, a decimal value of ``12288`` is the sum of ``4096`` + ``8192``, meaning the corresponding bits ``12` and ``13`` are ``1``, the rest a re ``0``. 
```

Comment:
I'm not sure how this is relevant--if a bitmask is used to extract a single bit, why do we need to add them?

Reply:
poked a bit more at it, and it seems like `0b1000` works fine :)

do you think that'd be easier to explain? And if so, would you like to update this guide?

Reply:
sure, but what more to explain?

the goal is to let the user know what a bitmask is and how to use it.

Reply:
I misunderstood what you were explaining here. Thanks for clarifying!